rest/oidc_api_test.go

```diff
@@ -0,0 +1,147 @@
+package rest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetOIDCCallbackURL(t *testing.T) {
+	var (
+		wellKnownPath           = "/.well-known/openid-configuration"
+		headerLocation          = "Location"
+		authProvider            = "provider"
+		authRedirectURI         = "redirect_uri"
+		providerNameGoogle      = "google"
+		providerNameSalesforce  = "salesforce"
+		clientIDGoogle          = "client.id.google"
+		clientIDSalesforce      = "client.id.salesforce"
+		validationKeySalesforce = "validation.key.salesforce"
+		validationKeyGoogle     = "validation.key.google"
+	)
+
+	// Mock OpenID Connect Provider Discovery Endpoints
+	var issuerGoogle, issuerSalesforce string
+	mux := http.NewServeMux()
+	mux.HandleFunc("/google/", func(res http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/google/.well-known/openid-configuration" {
+			http.NotFound(res, req)
+			return
+		}
```

Comment:
```suggestion
	mux.HandleFunc("/google"+wellKnownPath, func(res http.ResponseWriter, req *http.Request) {
```

Reply:
This change has been included in latest revision.